bokeh/plotting/helpers.py

```diff
@@ -513,6 +514,20 @@ def func(self, **kwargs):
         is_user_source = kwargs.get('source', None) is not None
         renderer_kws = _pop_renderer_args(kwargs)
         source = renderer_kws['data_source']
+        if not isinstance(source, ColumnDataSource):
+            try:
+                # try converting the soruce to ColumnDataSource
+                source = ColumnDataSource(source)
+            except ValueError as err:
+                msg = "Failed to auto-convert {curr_type} to ColumnDataSource.\n" \
+                      "Original error: {err}".format(
```

Comment:
slight preference for the string to be a single line with no \ (our linter max line length is 140 I think)